tests/sources/generate_diff.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+# Copyright 1996-2019 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate a list of modified files with respect to the parent branch.
+This script is used only by Travis."""
+import os
+import subprocess
+import sys
+
+branch = os.getenv('TRAVIS_BRANCH')  # branch targeted by the pull request
```

Comment:
I just read the documentation of this variable:
>for push builds, or builds not triggered by a pull request, this is the name of the branch.
for builds triggered by a pull request this is the name of the branch targeted by the pull request.
for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).

This means that if one day we will want to re-enable branch push test, this script will brake the test because it will test the diff with itself and therefore not test anything (without notifying us).